### PR TITLE
Activity log: Accept query parameters in activity data-layer

### PR DESCRIPTION
**Pull request by sirreal**
This PR adds `params` to the `activityLogRequest` action creator as well as the data-layer. This will allow us to take advantage of filtering provided by the endpoint like `date_start`, `date_end`, `number`, and eventually paging or other methods for dealing with management of large amounts of data.

## Testing
1. https://calypso.live/stats/activity?branch=update/activitylog-datalayer-accept-params
1. Dispatch this action:
   ```js
   dispatch( { type: 'ACTIVITY_LOG_REQUEST', params: { number: 3 }, siteId: state.ui.selectedSiteId } )
   ```
1. You should see that only 3 activities are loaded, and the network tab will display the provided params as well `number=3`.
1. Try passing other parameters and you should see them in the network requests.

### CODE REVIEW

File: client/state/activity-log/actions.js [hunk 1]
```diff
@@ -48,9 +48,10 @@ export function rewindActivateFailure( siteId ) {
 	};
 }
 
-export function activityLogRequest( siteId ) {
+export function activityLogRequest( siteId, params ) {
```
Comment (seear) [inline client/state/activity-log/actions.js]: Some jsdoc here with the accepted params could be useful if we know them.

File: client/state/data-layer/wpcom/sites/activity/index.js [hunk 2]
```diff
@@ -19,7 +19,10 @@ const activityLogRequest = ( { dispatch }, action ) => {
 		apiVersion: '1',
 		method: 'GET',
 		path: `/sites/${ action.siteId }/activity`,
-		query: { number: 1000 },
+		query: {
```
Comment (tyxla) [inline client/state/data-layer/wpcom/sites/activity/index.js]: I was surprised there are no tests for this bit at all - care to add some?

File: client/state/activity-log/actions.js (hunk 1 above)
Comment (sirreal) [inline client/state/activity-log/actions.js]: 👍  I added JSDocs in b109447, I'd appreciate a review.

Comment (seear) [inline client/state/activity-log/actions.js]: Looks great. I've seen dot notation used for nested params [elsewhere](https://github.com/Automattic/wp-calypso/blob/5c925841da2b5840d3b4fa3b7cc432a3f7876c80/client/lib/query-manager/index.js#L215) which is quite nice.

File: client/state/activity-log/actions.js [hunk 3]
```diff
@@ -48,9 +48,34 @@ export function rewindActivateFailure( siteId ) {
...
+ * @property {number} date_end   Filter activity before this date (utc microtime timestamp).
+ * @property {number} number     Maximum number of results to return.
+ */
+
+/**
+ * Requests activity from the API
+ *
+ * You may optionally pass an object of parameters for the query to this action.
+ * Some examples of parameters are:
+ *   {int}
```
Comment (seear) [inline client/state/activity-log/actions.js]: Underwhelming example :D

File: client/state/data-layer/wpcom/sites/activity/index.js (hunk 2 above)
Comment (sirreal) [inline client/state/data-layer/wpcom/sites/activity/index.js]: 👍  Added tests in db0072ab1ca38ab6d203c2c0222accd9811addce

File: client/state/activity-log/actions.js (hunk 3 above)
Comment (sirreal) [inline client/state/activity-log/actions.js]: 😅  Fixed that too `@see ActivityParams` to reference the preceding `@typedef`.

**Pull request merged**